### PR TITLE
Add file io

### PR DESCRIPTION
**Pull request by amberarrow**
@tweise Please take a look

### CODE REVIEW

File: tutorials/fileIO/pom.xml [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+  <artifactId>fileIO</artifactId>
+  <packaging>jar</packaging>
+
+  <!-- change these to the appropriate values -->
+  <name>File Input and Output</name>
+  <description>Simple application illustrating use of file input and output operators</description>
+
+  <properties>
+    <!-- change this if you desire to use a different version of DataTorrent -->
+    <apex.version>3.2.0-incubating</apex.version>
```
Comment (sandeepdeshmukh) [inline tutorials/fileIO/pom.xml]: Apex 3.3.0 is already released. Can this be also 3.3.0 with RTS 3.3.0 about to be released.

File: tutorials/fileIO/pom.xml [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+    </plugins>
+
+  </build>
+
+  <dependencies>
+    <!-- add your dependencies here -->
+    <dependency>
+      <groupId>com.datatorrent</groupId>
+      <artifactId>malhar-library</artifactId>
+      <version>3.1.1</version>
```
Comment (sandeepdeshmukh) [inline tutorials/fileIO/pom.xml]: malhar 3.3.1?

**Comment by amberarrow** [comment]
Updated core and malhar versions

File: tutorials/fileIO/src/main/java/com/example/fileIO/FileWriter.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+import com.datatorrent.lib.io.fs.AbstractFileOutputOperator;
+
+/**
+ * Write incoming line to output file
+ */
+public class FileWriter extends AbstractFileOutputOperator<String>
+{
+  private static final Logger LOG = LoggerFactory.getLogger(com.example.fileIO.FileWriter.class);
+  private static final String charsetName = "UTF-8";
```
Comment (tweise) [inline tutorials/fileIO/src/main/java/com/example/fileIO/FileWriter.java]: Variable name should be uppercase for static?

**Comment by tweise** [comment]
Would be nice to also have an example that shows partitioned output for a high throughput stream (that could be generated) along with rolling file option. That's typical for writing results of batch jobs.

**Comment by amberarrow** [comment]
Changed static variable names to uppercase; I'll add a different example for partitioned output.

**Comment by sandeepdeshmukh** [comment]
LGTM - Looks good to me .

**Pull request merged**